msb/analysis/timestepinterpolator.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+from msb.network.pubsub.types import Subscriber
+
+
+class PandasInterpolator:
+    """
+    An Iterator class that interpolates data points to a fixed rate.
+    """
+
+    def __init__(self, sub: Subscriber, freq_ms: float, sample_size: int = 100):
+        self.sub = sub
+        self.incoming: pd.DataFrame = None
+        self.outgoing: pd.DataFrame = None
+        self.resampling_frequency = freq_ms
+        self.sample_size = sample_size
+
+    def resample_loop(self) -> dict:
+        """
+        Generator loop that yields dicts at a fixed rate
+        """
+        while True:
+            self._listen_to_sub()
+            self._convert_timestamp()
+
+            # Reindex incoming timestamps
+            self.outgoing = self.incoming.resample(
+                f"{self.resampling_frequency}ms",
+                offset="0s",
+            ).mean()
+
+            # After resampling, convert from DateTimeIndex to Epoch as a field
+            # Index will not be returned in yield statement
+            self.outgoing["epoch"] = pd.to_numeric(self.outgoing.index) / 10**9
```

Comment:
Consider using one of the options shown in https://stackoverflow.com/a/54313505. Also try with `view` instead of `astype` as the comment states.
If performance is critical I would prefer the first option (I think this is basically what is done here), otherwise I would prefer the second option.

Reply:
This might also be its own method analogous to `_convert_timestamp`/`_epoch_to_datetime`.